File: Java/LeetCode/Find Triangular Sum of an Array.java
// https://leetcode.com/problems/find-triangular-sum-of-an-array

class Solution {
  
    public int triangularSum(int[] nums) {
        int n = nums.length;
        if (n == 1) {
            return nums[0];
        }

        int[] newNums = new int[n-1];
        for (int i = 0; i < n-1; ++i) {
            newNums[i] = (nums[i] + nums[i+1]) % 10;
        }

        nums = newNums;

        return triangularSum(nums);
    }
  
}

File: Kotlin/LeetCode/Find Triangular Sum of an Array.kt
// https://leetcode.com/problems/find-triangular-sum-of-an-array

class Solution {
  
    fun triangularSum(nums: IntArray): Int {
        val n = nums.size
        if (n == 1) {
            return nums[0]
        }

        val newNums = IntArray(n-1)
        for (i in 0 until n-1) {
            newNums[i] = (nums[i] + nums[i+1]) % 10
        }

        return triangularSum(newNums)
    }
    
}
